docs/practices/pull_requests.md

```diff
@@ -0,0 +1,79 @@
+# Pull request workflow
+
+Our team practices should encourage transparency and collaboration, and reviewing one another's work is a great way to ensure that we share context and knowledge, and improve the codebase.
+Our goals are to balance inclusive and transparent team practices with the reality that we are all asynchronous and don't want to pay a large coordination penalty for every change and decision.
+
+**All changes to a JupyterHub repository should happen via Pull Requests**.
+Any exceptions to this should be explicitly listed in team documentation.[^1]
+
+[^1]: The only exception we can think of is when pushing a commit to `main` during a release, but we'll leave this open-ended just in case there are other cases where this is useful.
+
+(prs:principles)=
+## Principles to follow
+
+- **Don't surprise people**. We pay a team penalty when somebody changes something that wouldn't be expected, especially if it is unwelcome.
+  Provide enough discussion and context sharing so that every team member understands the rationale for a change, even if they may not personally agree with it.
+  The more complex or impactful a change is, the more time should be given to discuss and include the perspective of others.
+- **Include others**. Actively invite participation and comments from others, especially if you think they are interested in the proposed change.
+  Go out of your way to make it easy for others to participate, and make your best effort at incorporating suggestions and feedback from others.
+  If a topic requires discussion, open an issue or a forum thread so that others can participate and align on an approach before diving into a code implementation.
+  We often save a lot of time by having design-level discussion before we get to the Pull Request.
+- **Link to discussions and context**. Almost any change should first be proposed in an issue, or in another space that is accessible to non-coders (like a forum post).
+  A Pull Request should link to (and optionally close) any other locations where discussion happened.
+  This gives each change more provenance and separates design from implementation.
```

Comment:
Hmm, maybe we can melt this one with the first bullet (see my suggestion about renaming it to `provide context`) to reduce any controversy?

Maybe something like:
```suggestion
- **Provide context**. Provide enough discussion and context sharing so that every team member understands the rationale for a change, even if they may not personally agree with it.
  A Pull Request should link to (and optionally close) any locations where discussion happened.
  This can give each change more provenance and separates design from implementation.
  The more complex or impactful a change is, the more time should be given to discuss and include the perspective of others. These discussions can happen in issues, pull requests or in another space that is accessible to non-coders (like a forum post). There aren't any strict rules about how to choose the discussion space, as this usually depends on the type, scope and size of change that's being proposed. So use your best judgement when choosing one.
```